components/skin/ui/src/main/resources/PhenoTips/Skin.xml

```diff
@@ -429,6 +429,130 @@ document.observe('xwiki:suggest:containerCreated', function(event) {
+        <prettyName>Code</prettyName>
+        <rows>20</rows>
+        <size>50</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </code>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>1</number>
+        <prettyName>Name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <parse>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>parse</name>
+        <number>4</number>
+        <prettyName>Parse content</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </parse>
+      <use>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>(function () {
+  var init = function() {
+    if ($('inline')) {
+      // Some form elements are still being modified after ""xwiki:dom:loaded" is fired.
+      // If the initial state of the form is recorded immediately, there will likely be
+      // differences with the state of the form when navigating the way even if the user
+      // hasn't actually changed any data.
+      // As a quick fix, wait two seconds before recording the "initial" state of the form.
+      // It's not ideal, but it will produce fewer false positives and no false negatives.
+      setTimeout(function(){window.__formData = $('inline').serialize(); }, 2000);
+      // Record the state of the form again once the form is saved.
+      document.observe('xwiki:document:saved', function(event) {
+         window.__formData = $('inline').serialize();
+      });
+      $$("input[name='action_save']").invoke('observe', 'click', function(event) {
+        window.__formData = $('inline').serialize();
+      });
+      // Upon navigating away, check if anything has changed in the form and warn the user
+      // if it did.
+      window.addEventListener('beforeunload', function(event) {
+        if (window.__formData != $('inline').serialize()) {
+          var message = "$services.localization.render('phenotips.navigation.beforeunload.warning')";
+          console &amp;&amp; console.log(message);
```

Comment:
I forgot to remove it. It was the only way to actually see the message was properly translated, since most browsers don't actually display the returned message.